python/lsst/ap/association/diaCalculation.py

```diff
@@ -0,0 +1,291 @@
+                raise ValueError("{} has an execution order less than the minimum for an catalogCalculation "
+                                 "plugin. Value {} : Minimum {}".format(*errorTuple))
+
+    @lsst.pipe.base.timeMethod
+    def run(self, diaObjectCat, diaSourceCat, updatedDiaObjectIds, filterName):
+        """The entry point for the DIA catalog calculation task.
+
+        Run method both updates the values in the diaObjectCat and appends
+        newly created DiaObjects to the catalog.
+
+        Parameters
+        ----------
+        diaObjectCat : `pandas.DataFrame`
+            DiaObjects to update values of and append new objects to. DataFrame
+            should be indexed on "diaObjectId"
+        diaSourceCat : `pandas.DataFrame`
+            DiaSources associated with the DiaObjects in diaObjectCat. DataFrame
+            should be indexed on `["diaObjectId", "filterName", "diaSourceId"]`
+        updatedDiaObjectIds : `numpy.ndarray`
+            Integer ids of the DiaObjects to update and create.
+        filterName : `str`
+            String name of the filter being processed.
+
+        Returns
+        -------
+        returnStruct : `lsst.pipe.base.Struct`
+        """
+        return self.callCompute(diaObjectCat,
+                                diaSourceCat,
+                                updatedDiaObjectIds,
+                                filterName)
+
+    def callCompute(self,
+                    diaObjectCat,
+                    diaSourceCat,
+                    updatedDiaObjectIds,
+                    filterName):
+        """Run each of the plugins on the catalog.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            The catalog on which the plugins will operate.
+        """
+
+        diaObjectUsed = pd.DataFrame(
+            False,
+            index=diaObjectCat.index,
+            columns=["used"])
+
+        updatedDiaObjects = []
+
+        for objId in updatedDiaObjectIds:
+            try:
+                updatedDiaObjDF = diaObjectCat.loc[objId]
+                updatedDiaObject = updatedDiaObjDF.to_dict()
+                updatedDiaObject["diaObjectId"] = objId
+                diaObjectUsed.loc[objId] = True
+            except KeyError:
+                updatedDiaObject = self._initialize_dia_object(objId)
+
+            # Sub-select diaSources associated with this diaObject.
+            objDiaSources = diaSourceCat.loc[objId]
+            # Currently needed as dataFrames loaded from sql do not currently
+            # map Null to NaN for custom queries. This can either stay here
+            # or move to dax_ppdb or ap_association.
+            objDiaSources.replace(to_replace=[None],
+                                  value=np.nan)
+
+            # Sub-select on diaSources observed in the current filter.
+            filterObjDiaSources = objDiaSources.loc[filterName]
```

Comment:
I'm mostly thinking that the majority of our pluggins will run a single band and in that case accessing this sub-dataframe for each pluggin would be an extra set computations that is necessary.

Reply:
Hmmm. Making the code more complicated here feels like a premature optimization. If (as we've discussed) we're effectively ignoring performance concerns at the moment because optimizing Pandas is too awkward, my instinct is to keep things simple here as well. We could add this logic when it's shown to actually make a difference.

...however, since this code already exists, I suppose we might as well keep it, with the caveat that we'll have to come back and think through how all of this performs later.